src/core/core.c

```diff
@@ -192,24 +211,49 @@ void core_preinit(const char *path)
 	const char *home;
 	char *str;
 	int len;
+	int use_xdg = 0;
 
 	if (irssi_dir == NULL) {
-		home = g_get_home_dir();
-		if (home == NULL)
-			home = ".";
-
-		irssi_dir = g_strdup_printf(IRSSI_DIR_FULL, home);
+		/* check if %XDG_CONFIG_HOME/irssi exists and use it */
+		char *dirp = g_build_filename(g_get_user_config_dir(), "irssi", NULL);
+		if (!g_file_test(dirp, G_FILE_TEST_EXISTS)) {
```

Comment:
could it be that this condition is inverted?

Reply:
That's funny. It is inverted.